packages/blocks/src/api/factory.js

```diff
@@ -47,7 +47,18 @@ export function createBlock( name, blockAttributes = {}, innerBlocks = [] ) {
 		if ( undefined !== value ) {
 			result[ key ] = value;
 		} else if ( schema.hasOwnProperty( 'default' ) ) {
-			result[ key ] = schema.default;
+			let defaultValue = schema.default;
+
+			// If the type of the attribute's default is a reference type,
+			// make a copy so that implementors cannot easily mutate the
+			// block type's default attributes.
+			if ( Array.isArray( defaultValue ) ) {
+				defaultValue = defaultValue.slice();
```

Comment:
You can also use ES6 syntax, not sure if it is faster though:
```js
defaultValue = [ ...defaultValue ];
defaultValue = { ...defaultValue }; 
```
but it's shorter for sure :)